pkg/kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go

```diff
@@ -491,29 +488,42 @@ func (c *muxStream) startRangeFeed(
 		return err
 	}
 
-	// mu must be held while marking this stream in flight (streams.Store) to
-	// synchronize with mux termination. When node mux terminates, it invokes
-	// c.closeWithRestart(), which marks this mux stream connection closed and
-	// restarts all active streams.  Thus, we must make sure that this streamID
-	// gets properly recorded even if mux go routine terminates right after the
-	// above sender.Send() succeeded.
-	c.streams.Store(streamID, unsafe.Pointer(stream))
+	c.mu.streams[streamID] = stream
```

Comment:
Could you add something about the race here, like:

> // Note that since we already called `Send(req)`, events from this stream might be showing up in the processing goroutines, and they need to be able to look up the stream. Since we still hold the mutex (and held it across `Send`), this will hold true. See: https://github.com/cockroachdb/cockroach/issues/97817.